src/expertsystem/reaction/_system_control.py

```diff
@@ -26,6 +30,103 @@
 ]
 
 
+def create_edge_properties(
+    particle: Particle,
+    spin_projection: Optional[float] = None,
+) -> GraphEdgePropertyMap:
+    edge_qn_mapping: Dict[str, Type[EdgeQuantumNumber]] = {
+        qn_name: qn_type
+        for qn_name, qn_type in EdgeQuantumNumbers.__dict__.items()
+        if not qn_name.startswith("__")
+    }  # Note using attr.fields does not work here because init=False
+    property_map: GraphEdgePropertyMap = {}
+    isospin = None
+    for qn_name, value in attr.asdict(particle).items():
```

Comment:
Isn't it better to iterate with `attr.fields`?

Reply:
See the code comment, the `attr.fields` are empty (i believe because of `init=False`)

Reply:
Ah yes overlooked that